testsuite/features/step_definitions/command_steps.rb

```diff
@@ -944,6 +941,28 @@
   $server.run(cmd)
 end
 
+When(/^I install "([^"]*)" product on the proxy$/) do |product|
+  out, = $proxy.run("zypper ref && zypper --non-interactive install --auto-agree-with-licenses --force-resolution -t product #{product}")
+  STDOUT.puts "Installed #{product} product: #{out}"
+end
+
+When(/^I install proxy pattern on the proxy$/) do
+  if $product == 'Uyuni'
+    pattern = 'uyuni_proxy'
+  else
+    pattern = 'suma_proxy'
+  end
+  cmd = "zypper --non-interactive install -t pattern #{pattern}"
+  $proxy.run(cmd, true, DEFAULT_TIMEOUT, 'root', [0, 100, 101, 102, 103, 106])
+end
+
+When(/^I let squid use avahi on the proxy$/) do
+  $proxy.run('cat >> /usr/share/rhn/proxy-template/squid.conf <<CONFIG
+dns_multicast_local on
+ignore_unknown_nameservers off
+CONFIG')
```

Comment:
That step will behave properly if we re-run the scenario containing it multiple times?

Reply:
Yes and no. It would be ugly, with duplicates. But I think the repeated lines would not prevent squid from working.

If you have a better solution, welcome.

The problem is that this series of PR is already inflating beyond reason as at the beginning it was supposed to address one simple file removed by accident :-) .

Reply:
And you say it :)
I think it would be better to address one by one in different PRs those changes. Instead of addressing too much in one.
I use to do the same, and I use to fail on that stone every time :)

Reply:
In any case, for the change, yes we can do it better, you did it bettee few days ago in another part of our code.
You checked with a grep if the change was already in place, before appending it, so I was inderectly suggesting the same ;p

Reply:
yes, you're right, we need to do it at some point

I think I'm reaching the point when I want to have this PR behind me :-) 

For now it's "good enough" as this is part of core == non idempotent

but yeah that's the solution

Reply:
https://github.com/uyuni-project/uyuni/blob/master/testsuite/features/step_definitions/common_steps.rb#L21

Reply:
done